src/application/service/GetExistingOrCreateNewUser.ts

```diff
@@ -0,0 +1,9 @@
+import { User } from '../../domain/model/user/User';
+import { v4 } from 'uuid';
+import { Email } from '../../domain/model/user/Email';
+
+export class GetExistingOrCreateNewUser {
+  public async execute(email: string) {
+    return new User(v4(), new Email(email));
```

Comment:
I think this class needs a big `TODO` :P

Reply:
yeah.. but let's not do a big bang PR and solve everything. One thing at a time :)

Reply:
Yep, my point was to add a `// TODO` as a comment. Try not to left things half done without marking them, as the probability for us to miss them in the future is higher